Sources/BraintreePayPalMessaging/BTPayPalMessagingView.swift

```diff
@@ -1,10 +1,149 @@
+
+    public weak var delegate: BTPayPalMessagingDelegate?
+
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializers
+
+    ///  Initializes a PayPal Messaging client.
+    /// - Parameter apiClient: The Braintree API client
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+
+        super.init(frame: .zero)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Method
+
+    /// Creates a `BTPayPalMessagingView` toto be displayed to promote offers such as Pay Later and PayPal Credit to customers.
+    /// - Parameter request: an optional `BTPayPalMessagingRequest`
+    public func createView(_ request: BTPayPalMessagingRequest = BTPayPalMessagingRequest()) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.delegate?.onError(self, error: error)
+                return
+            }
+            guard let configuration else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.fetchConfigurationFailed)
+                return
+            }
+
+            guard let clientID = configuration.json?["paypal"]["clientId"].asString() else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.payPalClientIDNotFound)
+                return
+            }
+
+            let messageData = PayPalMessageData(
+                clientID: clientID,
+                environment: configuration.environment == "production" ? .live : .sandbox,
+                amount: request.amount,
+                placement: request.placement?.placementRawValue,
+                offerType: request.offerType?.offerTypeRawValue
+            )
+
+            messageData.buyerCountry = request.buyerCountry
+
+            let messageConfig = PayPalMessageConfig(
+                data: messageData,
+                style: PayPalMessageStyle(
+                    logoType: request.logoType.logoTypeRawValue,
+                    color: request.color.messageColorRawValue,
+                    textAlignment: request.textAlignment.textAlignmentRawValue
+                )
+            )
+
+            let messageView = PayPalMessageView(config: messageConfig, stateDelegate: self, eventDelegate: self)
+            messageView.translatesAutoresizingMaskIntoConstraints = false
+            self.addSubview(messageView)
+
+            NSLayoutConstraint.activate([
+                messageView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+                messageView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+                messageView.topAnchor.constraint(equalTo: self.topAnchor),
+                messageView.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+            ])
+
+            return
+        }
+    }
+}
+
+// MARK: - UIViewRepresentable protocol conformance
+
+public extension BTPayPalMessagingView {
```

Comment:
This is our implementation for SwiftUI merchants. Have tested in the spike branch that this will work as expected for Swift UI merchants. Here is what the implementation looks like:
```swift
struct PayPalCreditMessagingSwiftUIView: View {

    let apiClient = BTAPIClient(authorization: "MY_AUTHORIZATION")!
    let request = BTPayPalMessagingRequest(
        amount: 2.0,
        offerType: .payLaterLongTerm,
        buyerCountry: "US",
        logoType: .primary,
        textAlignment: .center
    )

    var body: some View {
        BTPayPalMessagingView.Representable(apiClient: apiClient, request: request, delegate: messagingDelegate)
    }

    private var messagingDelegate: PayPalMessagingSwiftUIDelegate {
        PayPalMessagingSwiftUIDelegate()
    }
}

class PayPalMessagingSwiftUIDelegate: BTPayPalMessagingDelegate {

    func didSelect(_ payPalCreditMessageView: BTPayPalMessagingView) {
        print("didSelect fired")
    }

    func willApply(_ payPalCreditMessageView: BTPayPalMessagingView) {
        print("willApply fired")
    }

    func willAppear(_ payPalCreditMessageView: BTPayPalMessagingView) {
        print("willAppear fired")
    }

    func didAppear(_ payPalCreditMessageView: BTPayPalMessagingView) {
        print("didAppear fired")
    }

    func onError(_ payPalCreditMessageView: BTPayPalMessagingView, error: Error) {
        print("onError fired with \(error.localizedDescription)")
    }
}
```

Reply:
For a future ticket, but maybe we can add a separate Demo feature that is only in SwiftUI? Would be nice to also have reference to this in our demo somewhere

Reply:
Agree - it'd be great to have both in the same demo, but shouldn't be too hard for a future PR.